src/language_server_protocol.rs

```diff
@@ -258,6 +262,14 @@ impl LanguageClient {
             state.loggingLevel = loggingLevel;
             state.serverStderr = serverStderr;
             state.is_nvim = is_nvim;
+            if let Some(preferred_markup_kind) = preferred_markup_kind {
```

Comment:
Let's move this block outside `update`. For one, update closure should be fast to avoid locking global state for a long time. Secondly, if error occurred during deserialization, it would avoid the state partially synced.